packages/core/src/codewhisperer/util/zipUtil.ts

```diff
@@ -412,39 +410,56 @@ export class ZipUtil {
             return
         }
 
-        const sourceFiles = await collectFiles(
-            projectPaths,
-            (useCase === FeatureUseCase.TEST_GENERATION
+        const workspaceFolders = (
+            useCase === FeatureUseCase.TEST_GENERATION
                 ? [...(vscode.workspace.workspaceFolders ?? [])].sort(
                       (a, b) => b.uri.fsPath.length - a.uri.fsPath.length
                   )
-                : vscode.workspace.workspaceFolders) as CurrentWsFolders,
-            {
-                maxTotalSizeBytes: this.getProjectScanPayloadSizeLimitInBytes(),
-                excludePatterns:
-                    useCase === FeatureUseCase.TEST_GENERATION
-                        ? [...CodeWhispererConstants.testGenExcludePatterns, ...defaultExcludePatterns]
-                        : defaultExcludePatterns,
-            }
-        )
-        for (const file of sourceFiles) {
-            const projectName = path.basename(file.workspaceFolder.uri.fsPath)
-            const zipEntryPath = this.getZipEntryPath(projectName, file.relativeFilePath)
+                : vscode.workspace.workspaceFolders
+        ) as CurrentWsFolders
 
-            if (ZipConstants.knownBinaryFileExts.includes(path.extname(file.fileUri.fsPath))) {
-                if (useCase === FeatureUseCase.TEST_GENERATION) {
-                    continue
-                }
-                await this.processBinaryFile(zip, file.fileUri, zipEntryPath)
-            } else {
-                const isFileOpenAndDirty = this.isFileOpenAndDirty(file.fileUri)
-                const fileContent = isFileOpenAndDirty ? await this.getTextContent(file.fileUri) : file.fileContent
-                this.processTextFile(zip, file.fileUri, fileContent, languageCount, zipEntryPath)
+        const collectFilesOptions: CollectFilesOptions = {
+            maxTotalSizeBytes: this.getProjectScanPayloadSizeLimitInBytes(),
+            excludePatterns:
+                useCase === FeatureUseCase.TEST_GENERATION
```

Comment:
Do you think it would be hard to eventually get rid of these useCase conditions?

Reply:
Yea, this is what I am struggling with now. These enums are everywhere in `ZipUtil` and make it difficult to pull out general components.